File: lintcode/python/0793_intersection_of_arrays.py

# hash

class Solution:
    """
    @param arrs: the arrays
    @return: the number of the intersection of the arrays
    """
    def intersectionOfArrays(self, arrs):
        # write your code here
        count = {}
        # 记录每个数的出现次数
        for arr in arrs:
            for x in arr:
                if x not in count:
                    count[x] = 0
                count[x] += 1
        
        # 某个数出现次数等于数组个数，代表它在所有数组中都出现过
        result = 0
        for x in count.keys():
            if count[x] == len(arrs):
                result += 1
        return result

File: lintcode/golang/0793_intersection_of_arrays.go
//hash

/**
 * @param arrs: the arrays
 * @return: the number of the intersection of the arrays
 */
import "fmt"
func intersectionOfArrays (arrs [][]int) int {
    // write your code here
    n := len(arrs)
    fmt.Println(n)
    exists := make(map[int]int)
    for i := 0; i < n; i++ {
        fmt.Println("len: ", len(arrs[i]))
        if len(arrs[i]) == 1 && arrs[i][0] == 0 {
            continue
        }
        for _, v := range arrs[i] {
            fmt.Println(v)
            exists[v]++
        }
    }

    fmt.Println(exists)

    ret := 0
    for _, v := range exists {
        if v == n {
            ret++
        }
    }

    return ret
}

